### DISCUSSION TITLE
How to make only a button clickable?

### DISCUSSION THREAD

**Discussion by ivnsch** [react-dropzone/react-dropzone#1228] [Q&A] [answered]
How to implement the typical functionality "drag in the area or click button to browse files"?.

I followed the readme, but now the entire drag area is clickable. It should only be the button. If I move the `getRootProps` closer to the button, the area doesn't respond anymore to drag and drop.

```html
<div {...getRootProps({ className: "upload-container" })}>
    <div className="grey-190">Upload an image</div>
    <div className="upload-custom">
    <button className="file-custom secondary-button">Upload Image</button>
    <input
        {...getInputProps()}
        className="upload-input"
        type="file"
        accept="image/*"
    />
    </div>
    <div className="grey-190">or Drag and drop here</div>
</div>
```

**Comment by billyang1222** [comment]
see this https://react-dropzone.js.org/#src

**Comment by ivnsch** [comment]
@FireMapleLeaves how exactly does it help? sorry, I'm not working on this currently and can't experiment.

**Comment by vishalvannada** [comment, marked as answer]
There is documentation for this use case. Refer to https://react-dropzone.js.org/#section-opening-file-dialog-programmatically

You should set `noClick: true`, open file picker programmatically using the button.

**Comment by rolandjitsu** [comment]
@ivnsch the answer provided by @vishalvannada is what you're looking for.
